src/OpenCL/Extensions/Silk.NET.OpenCL.Extensions.ARM/Enums/SvmMemFlagsArm.gen.cs

```diff
@@ -11,7 +11,7 @@ namespace Silk.NET.OpenCL.Extensions.ARM
 {
     [System.Flags]
     [NativeName("Name", "cl_svm_mem_flags_arm")]
-    public enum SvmMemFlags : ulong
+    public enum SvmMemFlagsArm : ulong
```

Comment:
Why has the suffix appeared?

Reply:
The logic currently is "strip suffix is enum is unique" - helps with general naming quality but can lead to some glitches when adding more enums. In theory we could always strip the suffix as that is namespaced in C# but that doesn't really play that well with how typemaps currently work (being simple string-string)

Reply:
I guess this is a call upon you to decide which is more usable. I personally would prefer stripping the extension prefix from enums in extension packages, as this is how C# works, but this isn't a precedent set elsewhere in the library as everywhere else puts enums and structs in the core packages.

Reply:
OK, interesting, so only enums are separated?
I guess this relates to a breaking change above.

Thing is, this entire story started with OpenCL enums being variably sized, so using "one size fits all" CLEnum won't fly (because it's not one size and it doesn't fit all).
So enum members must be correctly typed to get correct struct sizes, having those enums in separate assemblies doesn't help.
On top of that, having extension-specific structs in the core package serves little purpose as you'd want to reference the extension package to use them anyway.
Or am I missing something here?

And, last but not the least, OpenCL extensions are broken in Silk as of now... GetProcAddress cannot be used in OpenCL, and I'm really surprised there's not a single issue open on the topic.

Reply:
> OK, interesting, so only enums are separated?

Not sure what you mean by this, but in other APIs the features and extensions have no knowledge of groups thus we just bundle them all in the core assembly.

> Thing is, this entire story started with OpenCL enums being variably sized, so using "one size fits all" CLEnum won't fly (because it's not one size and it doesn't fit all).

Wasn't denying this. Not sure how this links back to my original prefix question though.

 > So enum members must be correctly typed to get correct struct sizes, having those enums in separate assemblies doesn't help.

Not sure how the assembly of residence and the enum size are linked, nor how this links back to my original prefix question.

> On top of that, having extension-specific structs in the core package serves little purpose as you'd want to reference the extension package to use them anyway.

Agree, but this is a breaking change we can't justify. Breaking changes we can justify are changes where the upstream specification has changed the way they represent the API or otherwise changed the spec, meaning that the breaking change occurs regardless of changes to BuildTools. Breaking changes we can't justify is where we change our interpretation of the specification, meaning that a breaking change occurs regardless of changes to the upstream specification. We plan to fix things like this in 3.0, as Vulkan has the same problem wherein extension structs are in the core packages but we aren't doing anything about it as we can't justify the breaking change outside of a major release.

Reply:
Well, those are related if you start tracking types of enums and actually strongly type enum members.
Anyway, rolled that back.

Enum prefix removal is intended to mimic a more "human-like" naming style, so I guess it's better to have it enabled...